presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -481,6 +562,172 @@ public static void multiply(Slice left, Slice right, Slice result)
         pack(result, (int) z0, (int) z1, (int) z2, (int) z3, leftNegative != rightNegative);
     }
 
+    public static void multiply(Slice left, long right, Slice result)
+    {
+        checkArgument(result.length() == NUMBER_OF_LONGS * Long.BYTES);
+
+        long l0 = toUnsignedLong(getInt(left, 0));
+        long l1 = toUnsignedLong(getInt(left, 1));
+        long l2 = toUnsignedLong(getInt(left, 2));
```

Comment:
no explanation how it all works, it's very hard to decipher it from looking at code